GC Classic 14.1.1: failed to compile after applying a custom mechanism

### Name and Institution (Required)

Name: Aoxing Zhang
Institution: Southern University of Science and Technology

### Confirm you have reviewed the following documentation

- [ x ] [Support guidelines](https://geos-chem.readthedocs.io/en/stable/reference/SUPPORT.html)
- [ x ] [User manuals](https://geos-chem.readthedocs.io/en/stable/geos-chem-shared-docs/supplemental-guides/related-docs.html)
- [ x ] [Debugging GEOS-Chem and HEMCO errors](https://geos-chem.readthedocs.io/en/stable/geos-chem-shared-docs/supplemental-guides/debug-guide.html)
-  [ x ] [Current and past Github issues](https://github.com/geoschem/geos-chem/issues)

### Description of your issue or question

Please provide as much detail as possible. Always include the GEOS-Chem version number and any relevant configuration and log files.

Dear GC support team,

I am using the GC Classic 14.1.1 and tried to update some changes for the chemical mechanisms. I followed the guidance of "Update chemical mechanisms with KPP" (https://geos-chem.readthedocs.io/en/stable/geos-chem-shared-docs/supplemental-guides/using-kpp-with-gc.html?highlight=add%20new%20reaction), changed some reactions (but not adding species) and rebuild KPP, it seemed work:

`./build_mechanism.sh custom`
```
This is KPP-2.5.0.

KPP is parsing the equation file.
KPP is computing Jacobian sparsity structure.
KPP is starting the code generation.
KPP is initializing the code generation.
KPP is generating the monitor data:
    - gckpp_Monitor
KPP is generating the utility data:
    - gckpp_Util
KPP is generating the global declarations:
    - gckpp_Main
KPP is generating the ODE function:
    - gckpp_Function
KPP is generating the ODE Jacobian:
    - gckpp_Jacobian
    - gckpp_JacobianSP
KPP is generating the linear algebra routines:
    - gckpp_LinearAlgebra
KPP is generating the utility functions:
    - gckpp_Util
KPP is generating the rate laws:
    - gckpp_Rates
KPP is generating the parameters:
    - gckpp_Parameters
KPP is generating the global data:
    - gckpp_Global
KPP is generating the driver from none.F90:
    - gckpp_Main
KPP is starting the code post-processing.

KPP has succesfully created the model "gckpp".

Reactivity consists of 189 reactions
Written to gckpp_Util.F90
```
Then I go to my run dic to rebuild this using the following commands:
```
cmake ../CodeDir -DMECH=custom -DRUNDIR=..
make -j
```
The cmake part works, ended by:
```
=================================================================
GEOS-Chem 14.1.1 (science codebase)
Current status: 14.1.0-134-g462e421ac.dirty
=================================================================
Creating /work/ese-zhangax/data/GC14.1.1/rundic/GEOSFP_4x5_GLYXupdate/CodeDir/src/GEOS-Chem/Interfaces/GCClassic/gc_classic_version.H
-- Configuring done (2.5s)
-- Generating done (0.3s)
-- Build files have been written to: /work/ese-zhangax/data/GC14.1.1/rundic/GEOSFP_4x5_GLYXupdate/build

```

but make -j failed, giving the following errors:


```
[ 86%] Building Fortran object src/GEOS-Chem/GeosCore/CMakeFiles/GeosCore.dir/emissions_mod.F90.o
[ 86%] Building Fortran object src/GEOS-Chem/GeosCore/CMakeFiles/GeosCore.dir/chemistry_mod.F90.o
[ 87%] Building Fortran object src/GEOS-Chem/GeosCore/CMakeFiles/GeosCore.dir/gc_environment_mod.F90.o
[ 89%] Building Fortran object src/GEOS-Chem/GeosCore/CMakeFiles/GeosCore.dir/cleanup.F90.o
[ 89%] Linking Fortran static library libGeosCore.a
[ 98%] Built target GeosCore
[ 98%] Building Fortran object src/CMakeFiles/gcclassic.dir/GEOS-Chem/Interfaces/GCClassic/main.F90.o
[100%] Linking Fortran executable ../bin/gcclassic
GEOS-Chem/GeosCore/libGeosCore.a(fullchem_mod.F90.o): In function `fullchem_mod_mp_do_fullchem_':
fullchem_mod.F90:(.text+0x65c3): undefined reference to `fullchem_autoreducefuncs_mp_fullchem_ar_keephalogensactive_'
fullchem_mod.F90:(.text+0x802a): undefined reference to `fullchem_autoreducefuncs_mp_fullchem_ar_setkeepactive_'
fullchem_mod.F90:(.text+0x91b2): undefined reference to `fullchem_autoreducefuncs_mp_fullchem_ar_setintegratoroptions_'
fullchem_mod.F90:(.text+0xe484): undefined reference to `fullchem_autoreducefuncs_mp_fullchem_ar_updatekppdiags_'
GEOS-Chem/GeosCore/libGeosCore.a(carbon_gases_mod.F90.o): In function `carbon_gases_mod_mp_chem_carbon_gases_':
carbon_gases_mod.F90:(.text+0x60ec): undefined reference to `carbon_funcs_mp_carbon_convertkgtomoleccm3_'
carbon_gases_mod.F90:(.text+0x62ee): undefined reference to `carbon_funcs_mp_carbon_computerateconstants_'
carbon_gases_mod.F90:(.text+0x6720): undefined reference to `carbon_funcs_mp_carbon_convertmoleccm3tokg_'
carbon_gases_mod.F90:(.text+0x7064): undefined reference to `carbon_funcs_mp_carbon_get_co2fromoh_flux_'
carbon_gases_mod.F90:(.text+0x7124): undefined reference to `carbon_funcs_mp_carbon_get_cofromnmvoc_flux_'
carbon_gases_mod.F90:(.text+0x719e): undefined reference to `carbon_funcs_mp_carbon_get_cofromch4_flux_'
make[2]: *** [bin/gcclassic] Error 1
make[1]: *** [src/CMakeFiles/gcclassic.dir/all] Error 2
make: *** [all] Error 2
```

There is no error reports before, so I'm not sure what these undefined references refers to...

Do you have any suggestions, or what procedure might I miss when updating the chemical mechanism?

Thanks,
Aoxing

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,6 +4,10 @@ This file documents all notable changes to the GEOS-Chem repository starting in 
 
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [Unreleased]
+### Changed
+- The `fullchem` mechanism must now be built with KPP 3.0.0 or later
+
 ## [Unreleased 14.2.0]
 ### Added
 - Added a printout of GEOS-Chem species and indices

--- a/KPP/fullchem/CHANGELOG_fullchem.md
+++ b/KPP/fullchem/CHANGELOG_fullchem.md
@@ -4,6 +4,10 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 
 # Mechanism history
 
+## [Unreleased]
+### Changed
+- The `fullchem` mechanism must now be built with KPP 3.0.0 or later
+
 ## [Unreleased 14.2.0]
 ### Added
 - Added lumped furan chemistry following Carter2020

--- a/KPP/fullchem/fullchem.kpp
+++ b/KPP/fullchem/fullchem.kpp
@@ -1,4 +1,4 @@
-#MINVERSION   2.5.0                  { Need this version of KPP or later          }
+#MINVERSION   3.0.0                  { Need this version of KPP or later          }
 #INTEGRATOR   rosenbrock_autoreduce  { Use Rosenbrock integration method          }
 #AUTOREDUCE   on                     { ... with autoreduce enabled but optional   }
 #LANGUAGE     Fortran90              { Generate solver code in Fortran90 ...      }

